src/exchange/swap/hooks/usePickDefaultAccount.ts

```diff
@@ -1,5 +1,6 @@
 import { useEffect } from "react";
 import { Account, TokenAccount } from "../../../types";
+import { BigNumber } from "bignumber.js";
 
 // Pick a default source account if none are selected.
 export const usePickDefaultAccount = (
```

Comment:
the reeduce logic seems a bit overcomplicated to me if you're just trying to filter and sort the accounts :shrug: 

```js
const highestbalanceAccount = accounts.filter(acc => acc.balance.gt(0))
                               .sort((a,b) => a.balance.gt(b.balance)))
                               .find(Boolean);
```

should do the trick in that case